src/test/java/org/broadinstitute/hellbender/utils/downsampling/ReservoirDownsamplerUnitTest.java

```diff
@@ -124,6 +127,41 @@ public void testSignalNoMoreReadsBefore() throws Exception {
         rd.submit(r2);
     }
 
+    @Test
+    public void testDownsampleByMappingQuality() {
+        Utils.resetRandomGenerator();
+        final ReservoirDownsampler rd = new ReservoirDownsampler(10, false, true, Integer.MAX_VALUE);
+        final List<GATKRead> reads = IntStream.range(0, 100)
+                .mapToObj(n -> {
+                    final GATKRead read = ArtificialReadUtils.createArtificialRead("100M");
+                    read.setMappingQuality(n);
+                    return read;
+                }).collect(Collectors.toList());
+        Collections.shuffle(reads, Utils.getRandomGenerator());
+        rd.submit(reads);
+        rd.signalEndOfInput();
+        final List<GATKRead> downsampledReads = rd.consumeFinalizedItems();
+        Assert.assertEquals(downsampledReads.size(), 10);
+        final double averageMappingQuality = downsampledReads.stream()
+                .mapToInt(GATKRead::getMappingQuality).average().getAsDouble();
+        Assert.assertTrue(averageMappingQuality > 70);
+    }
+
+    @Test
+    public void testDepthToIgnoreLocus() {
+        Utils.resetRandomGenerator();
+        final ReservoirDownsampler rd100 = new ReservoirDownsampler(10, false, false, 100);
```

Comment:
done and did something in the same spirit.  Instead I extracted `numReads = 100` and set `depthToIgnore` to `numReads` and `numReads + 1`.